packages/core/src/routes/connector.ts

```diff
@@ -0,0 +1,51 @@
+import { Connectors } from '@logto/schemas';
+import { object, string } from 'zod';
+
+import { validateConfig } from '@/connectors/utilities';
+import koaGuard from '@/middleware/koa-guard';
+import { findAllConnectors, findConnectorById, updateConnector } from '@/queries/connector';
+
+import { AuthedRouter } from './types';
+
+export default function connectorRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/connectors', async (ctx, next) => {
+    ctx.body = await findAllConnectors();
+
+    return next();
+  });
+
+  router.get(
+    '/connectors/:id',
+    koaGuard({ params: object({ id: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+      ctx.body = await findConnectorById(id);
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/connectors/:id',
+    koaGuard({
+      params: object({ id: string().min(1) }),
+      body: Connectors.guard.omit({ id: true, type: true, createdAt: true }).partial(),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+        body: { enabled, config },
```

Comment:
it would be better to create `PATCH /connectors/:id/enabled`, there are several benifits:

1. The front end (console) may want to change `enabled` only in a single request (with a switch in UI).
2. Changing `enabled` may need to `validateConfig`, in a seperate route would make the code more clean.

Reply:
please left the conversation unresolved until the original commenter resolves / confirms.

Reply:
the latest version still doesn't fit. `enabled` should be totally isolated into another route.

Reply:
per our current connector schema design, `enabled` field and `config` field should be put in two different routes?
a.k.a. `config` should also be a totally isolated route?

Reply:
`config` can still remain in this route, because is just a simple update, depends on you.

Reply:
@wangsijie i think this is sill not resolved since the route logic looks not good, @darcyYe would you mind refactoring this part?